indico/modules/events/registration/client/js/form/fields/TimetableSessions.jsx

```diff
@@ -0,0 +1,199 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import React, {useState, useEffect} from 'react';
+import {useSelector} from 'react-redux';
+import {Form} from 'semantic-ui-react';
+
+import {FinalCheckbox, FinalRadio, FinalField, FinalInput, validators as v} from 'indico/react/forms';
+
+import {Translate} from 'indico/react/i18n';
+
+import timeTableURL from 'indico-url:event_registration.api_event_timetable';
+import '../../../styles/regform.module.scss';
+import {getStaticData} from '../selectors';
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+import PropTypes from 'prop-types';
+
+function TimetableSessionsComponent({htmlId, htmlName, disabled, title, isRequired, ...props}) {
+  const [sessionData, setSessionData] = useState([]);
+  const {eventId} = useSelector(getStaticData);
+
+  useEffect(() => {
+    const fetchSessionData = async () => {
+      try {
+        const sessionsResponse = await indicoAxios.get(timeTableURL({event_id: eventId}));
+        setSessionData(sessionsResponse.data);
+      } catch (e) {
+        handleAxiosError(e);
+        return;
+      }
+    };
+    fetchSessionData(eventId);
+  }, [eventId]);
```

Comment:
`useIndicoAxios` is your friend for fetching things once

Reply:
You know why this might be happening?
<img width="899" alt="Screenshot 2024-02-15 at 13 18 37" src="https://github.com/indico/indico/assets/11741165/853d42d1-3c40-4b77-a811-5ede01fdb197">

Reply:
"this"?

Check your devtools to see if a request has been sent and what has been returned by it.

Reply:
No, there's no request sent, that's why it returns null and that's why I am asking, because I have seen in other examples this exact syntax and my previous endpoint was working with useEffect, but now it is not, and I don't understand the reason.

I guess I am using `useIndicoAxios` wrongly, but I don't know what could be wrong, maybe I need to import something else

Reply:
The code is pushed btw

Reply:
Check your devtools. It breaks because during the first render you already use `sessionData.map(...)` but at that point `sessionData` is still `null`. And any uncaught exception in react caused the whole react 'app' to get unmounted, so the request never got sent.

With this small change it'll work:

```patch
diff --git a/indico/modules/events/registration/client/js/form/fields/TimetableSessions.jsx b/indico/modules/events/registration/client/js/form/fields/TimetableSessions.jsx
index 3d9ddf8f2c..f2877514da 100644
--- a/indico/modules/events/registration/client/js/form/fields/TimetableSessions.jsx
+++ b/indico/modules/events/registration/client/js/form/fields/TimetableSessions.jsx
@@ -40,7 +40,7 @@ function TimetableSessionsComponent({htmlId, htmlName, disabled, title, isRequir

   return (
     <Form.Group styleName="accompanyingpersons-field" name={htmlName} id={htmlId}>
-      {sessionData.map((day, index) => (
+      {sessionData?.map((day, index) => (
         <SessionBlock
           key={index}
           sessionBlock={day}
```

(of course you should ideally show a loading indicator etc while it's loading, but for testing that's enough)